src/init.gradle

```diff
@@ -88,6 +90,44 @@ rootProject {
         }
     }
 
+    def cycloneDxBOMTask = tasks.register("generateCycloneDxBom", org.cyclonedx.gradle.CycloneDxTask) {
+        mustRunAfter(dependencyTree)
+        doFirst {
+            configurations {
+                customDependencies
+            }
+
+            def dependencyTreeFile = rootProject.layout.buildDirectory.file("licenses/tree_all.txt")
+            def lines = dependencyTreeFile.get().asFile.readLines()
+
+            // Add dependencies from top level 'tree_all.txt' to the custom configuration
+            for (String line : lines) {
+                def dependency = line.trim()
+                if (dependency == "" || !dependency.contains(".")) { break }
+                dependencies {
+                    customDependencies dependency
+                }
+            }
+        }
+
+        schemaVersion = "1.6"
+        outputFormat = "json"
+        includeConfigs = ["customDependencies"]
+        destination = file("build/reports/sbom")
+        outputName = "unfiltered-sbom"
+    }
+
+    def filterSbomTask = tasks.register("filterSBOM", FilterSBOM) {
+        mustRunAfter(cycloneDxBOMTask)
```

Comment:
Explicit dependency is very likely to be an accidental dependency. See https://melix.github.io/blog/2021/10/gradle-quickie-dependson.html

You should wire the task input properly instead.